mobile-app/app/screens/AppNavigator/screens/Balances/hooks/TokenPrice.ts

```diff
@@ -50,19 +58,24 @@ export function useTokenPrice (): DexTokenPrice {
       const usdTokenB = getTokenPrice(pair.data.tokenB.symbol, tokenBAmount.toFixed(8))
       return usdTokenA.plus(usdTokenB)
     }
+    const key = `WALLET.${network}.${blockCount ?? 0}.TOKEN_PRICE_${symbol}`
+    const result = CacheApi.get(key)
+    if (result !== undefined) {
+      return new BigNumber(result).multipliedBy(amount)
+    }
     // active price for walletTokens based on USDT
     const arbitraryPoolPair = getArbitraryPoolPair(symbol, 'USDT')
 
-    if (symbol === 'USDT') {
-      return new BigNumber(amount)
-    } else if (arbitraryPoolPair.length > 0) {
-      const { estimated } = calculatePriceRates(symbol, arbitraryPoolPair, amount)
+    if (arbitraryPoolPair.length > 0) {
```

Comment:
```suggestion
    if (arbitraryPoolPair?.length > 0) {
```

Reply:
` const arbitraryPoolPair = getArbitraryPoolPair(symbol, 'USDT')` this will always return array in return so its not required to use `arbitraryPoolPair?.length`